tensorboard/plugins/graph/metadata.py

```diff
@@ -24,3 +24,10 @@
 # graph Summaries.
 # See `graphs_plugin.py` for details.
 PLUGIN_NAME = "graphs"
+
+# In the context of the data provider interface, tag name given to a
+# graph that logically applies at the run level. No other types of
+# graphs are currently supported; when added, they'll have type-specific
+# name scopes (like `"__op_graph__/foo"`, `"__conceptual_graph__/bar"`)
```

Comment:
Note that if `__run_graph__/__op_graph__` refers to the op graph at run
level, and `foo/__op_graph__` refers to the op graph with user-provided
name `foo`, then there’s still a potential collision if the user chooses
to provide the name `__run_graph__`.

> the run-level graphs (which could be op or conceptual or both)

Run-level graphs are only ever op graphs. By “run-level graphs” we mean
“those graphs attached to a run by virtue of being stored on the
`graph_def` field of an `Event` proto, with no corresponding tag”. The
`graph_def` field is always interpreted as a `GraphDef`, never a Keras
model or anything like that. New-style op graphs, conceptual graphs,
profile graphs, etc. are all written as summaries that _do_ have tag
names.

The writer shuold of course make sure not to write conflicting data for
the same tag name, but run graphs don’t natively _have_ a tag name, so
we have to synthesize one—thence the potential for collision.

Anyway, yes, we can of coruse decide on the naming for the other parts
of the hierarchy when it’s actually needed.

Reply:
OK, in that case can we call the run graph `__op_graph/`, i.e., an op graph where the tag name is the empty string (which, I *hope* but don't recall, we should treat as equivalent to `None`)?

Barring that, maybe amend this code comment with (more or less) the second paragraph of your response here?

Reply:
Sure; amended the comment accordingly.

It seems confusing to me to call the run graph `__op_graph/`, given that
the plugin implementation draws an explicit distinction between run
graphs and op graphs. For instance, the `/info` route will return
`"run_graph": True` and not `op_graph: "True"` for the run graph. It
might or might not be reasonable to restructure the plugin to treat
things more homogeneously, but again this commit is just a pointed bug
fix and should not make any such changes. So I’ve kept `__run_graph__`
for now, because that’s consistent with all the other code.

I’m not exactly sure what you mean by “an op graph where the tag name is
the empty string (which, I _hope_ but don't recall, we should treat as
equivalent to `None`)”. Whether a run is an op graph or not is properly
signaled out of band by `"run_graph"`/`"op_graph"` keys, not inferred
from whether a tag name is empty, and tag names cannot be `None`.

Reply:
OK, thanks.  Yep I was basically asking about more homogeneous treatment of run_graph vs op_graph, but I accept the argument that that would be out of scope here.